dart/dynamics/BodyNode.h

```diff
@@ -613,6 +591,60 @@ class BodyNode :
   template <class NodeType, typename ...Args>
   NodeType* createNode(Args&&... args);
 
+  /// Create an ShapeNode attached to this BodyNode. Pass an
+  /// ShapeNode::Properties argument into this function.
+  template <class ShapeNodeProperties>
+  ShapeNode* createShapeNode(const ShapeNodeProperties& properties);
+
+  /// Create an ShapeNode with the specified name
+  ShapeNode* createShapeNode(
+      const ShapePtr& shape, const std::string& name = "ShapeNode");
+
+  /// Create an ShapeNode with the specified name
+  ShapeNode* createShapeNode(const ShapePtr& shape, const char* name);
```

Comment:
I think the way it works is that it will attempt to use a templated function instead of performing an implicit conversion, but then the templated function fails because it wasn't meant for the case of a ``const char*``.